### ISSUE TITLE
Exception while evaluating QuantumKernel on a QASM simulator when duplicate entries in the dataset

### ISSUE THREAD

**Issue by adekusar-drl** [Qiskit/qiskit-machine-learning#336]
### Environment

- **Qiskit Machine Learning version**: latest sources
- **Python version**: all
- **Operating system**: all
 

### What is happening?

Exception is raised when a quantum kernel is evaluated for a dataset with duplicate entries on a qasm simulator:
```
FAILURE: Can not get job id, Resubmit the qobj to get job id. Error: max() arg is an empty sequence 
.... This line appears many times ...
FAILURE: Can not get job id, Resubmit the qobj to get job id. Error: max() arg is an empty sequence 
Traceback (most recent call last):
  File ".../qiskit-machine-learning/_sandbox/qk_bug.py", line 14, in <module>
    qasm_kernel.evaluate(x_vec=train_features)
  File ".../qiskit-machine-learning/qiskit_machine_learning/kernels/quantum_kernel.py", line 535, in evaluate
    results = self._quantum_instance.execute(circuits, had_transpiled=True)
  File ".../envs/dev-qml/lib/site-packages/qiskit/utils/quantum_instance.py", line 811, in execute
    if circuit_job
  File ".../envs/dev-qml/lib/site-packages/qiskit/utils/run_circuits.py", line 530, in run_circuits
    max_job_retries=max_job_retries,
  File ".../envs/dev-qml/lib/site-packages/qiskit/utils/run_circuits.py", line 710, in _safe_submit_circuits
    raise QiskitError("Max retry limit reached. Failed to submit the qobj correctly")
qiskit.exceptions.QiskitError: 'Max retry limit reached. Failed to submit the qobj correctly'
```

### How can we reproduce the issue?

Run the script:

```python
import numpy as np
from qiskit import Aer
from qiskit.circuit.library import ZZFeatureMap
from qiskit.utils import QuantumInstance

from qiskit_machine_learning.kernels import QuantumKernel

qasm_backend = QuantumInstance(Aer.get_backend("qasm_simulator"))   # statevector_simulator

feature_map = ZZFeatureMap(feature_dimension=2)
qasm_kernel = QuantumKernel(feature_map=feature_map, quantum_instance=qasm_backend)

train_features = np.array([[1, 2], [1, 2]])
qasm_kernel.evaluate(x_vec=train_features)
```

If you replace `qasm_simulator` with `statevector_simulator` no exception is raised.

### What should happen?

No exception should be raised.

### Any suggestions?

This problem happens due to this check: https://github.com/Qiskit/qiskit-machine-learning/blob/main/qiskit_machine_learning/kernels/quantum_kernel.py#L524-L526

There's a check if two samples are identical and if this is true these samples are not used for quantum kernel estimation and since in the original dataset there are just two identical samples, nothing can be estimated and an empty list is passed to `QuantumInstance` for execution and it fails.

May be a warning should be raised.

**Comment by pranavkubc** [comment]
Hi,
I was running into a similar issue with statevector simulator. I resolved it by changing the "max_job_retries" attribute in QuantumInstance.

`quantum_instance = QuantumInstance(backend, shots = 1024, backend_options=backend_options,max_job_retries=-1)`

you can change the number to any positive integer that you want. Default is 50. You can change into -1 if you want infinte max_job_retries

**Comment by adekusar-drl** [comment]
Closed in #432
